### PR TITLE
LPS-147481

### PR DESCRIPTION
**Pull request by eliasantosf**
Ticket: [LPS-147481](https://issues.liferay.com/browse/LPS-147481)
Story: [LPS-139394](https://issues.liferay.com/browse/LPS-139394)

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/AssetCreator.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import {DiagramBuilderContext} from '../../../../../DiagramBuilderContext';
+
+const AssetCreator = () => {
+	const {setSelectedItem} = useContext(DiagramBuilderContext);
+	useEffect(() => {
+		setSelectedItem((previousValue) => ({
+			...previousValue,
+			data: {
+				...previousValue.data,
+				taskTimes: {
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/AssetCreator.js:23]: @eliasantosf I believe this should be `taskTimers`

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/SelectReassignment.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * distribution rights of the Software.
+ */
+
+import ClayForm, {ClaySelect} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React from 'react';
+
+import SidebarPanel from '../../../SidebarPanel';
+
+const options = [
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/SelectReassignment.js:18]: @eliasantosf You could move this array to an external file and import it here and in the `SelectAssignment` file, since they use the same options.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js [hunk 3]
```diff
@@ -9,58 +9,59 @@
...
+	const [sections, setSections] = useState([{identifier: `${Date.now()}-0`}]);
+	const ReassignmentSectionComponent = assignmentSectionComponents[section];
 
-			return newSections;
-		});
-	};
+	const [scriptSections] = useState([{identifier: `${Date.now()}-0`}]);
 
-	return reassignmentSections.map(({identifier}) => {
+	return scriptSections.map(({identifier}) => {
```
Comment (naraandrade) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js:41]: @eliasantosf I don't understand why this state is named `scriptSections` instead of `reassignmentSections`

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by naraandrade** [comment]
@eliasantosf In addtition to the comments above, I noticed that the delete in each section of User and RoleType is not working and the application is crashing. The "Add Source Code" button on ScriptedAssignment is not working as well.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/SelectReassignment.js (hunk 2 above)
Comment (eliasantosf) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/SelectReassignment.js:18]: I suggest opening a technical debt in the future for this change, because the options are not exactly the same, the last element of the Array is different. Therefore, changing only the last element of this Array at the moment will bring more difficulties to deliveries, as other functions must be used to change the Array value in certain sections.

**Comment by cgmonte** [comment]
Just started reviewing :)

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js [hunk 4]
```diff
@@ -36,9 +36,9 @@ const ActionTypeReassignment = (props) => {
 	const [sections, setSections] = useState([{identifier: `${Date.now()}-0`}]);
 	const ReassignmentSectionComponent = assignmentSectionComponents[section];
 
-	const [scriptSections] = useState([{identifier: `${Date.now()}-0`}]);
+	const [reassignmentSections] = useState([{identifier: `${Date.now()}-0`}]);
```
Comment (cgmonte) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js:39]: Please see the next comment, as this state is not necessary and should be removed.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js [hunk 5]
```diff
@@ -36,9 +36,9 @@ const ActionTypeReassignment = (props) => {
...
 
-	return scriptSections.map(({identifier}) => {
+	return reassignmentSections.map(({identifier}) => {
```
Comment (cgmonte) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js:41]: Please remove both return and map, SelectReassignment should be returned directly. Map only occurs in sections.map.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js [hunk 6]
```diff
@@ -36,9 +36,9 @@ const ActionTypeReassignment = (props) => {
...
 		return (
 			<div key={`section-${identifier}`}>
```
Comment (cgmonte) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js:43]: This div should be replace with fragments.

**Review by cgmonte — CHANGES_REQUESTED** [review]
This commit (LPS-147481 Adjust in select-reassignment folder files) should be squashed with LPS-147481 Creation and definition of select-reassignment folder

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 7]
```diff
@@ -3405,6 +3405,7 @@ commerce.guest.checkout.authentication.commerce.health.status.description=Check
 commerce.guest.checkout.authentication.commerce.health.status.key=Guest Checkout Authentication
 common-items=Common Items
 common-name=Common Name
+common-styles=Common Styles
```
Comment (cgmonte) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:3408]: Is this related to ticket LPS-147481?

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 8]
```diff
@@ -9169,8 +9170,6 @@ miller-columns=Miller Columns
 millimeter=Millimeter
 million-abbreviation=M
 mime-type=MIME Type
-mime-type-size-limit-configuration-name=Size Limits by Mime Type
```
Comment (cgmonte) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:9172]: Is this related to ticket LPS-147481?

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 9]
```diff
@@ -9169,8 +9170,6 @@ miller-columns=Miller Columns
...
-mime-type-size-limit-name=Mime Type Size Limit
```
Comment (cgmonte) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:9173]: Is this related to ticket LPS-147481?

**Review by cgmonte — CHANGES_REQUESTED** [review]
please squash this commit (LPS-147481 Update PropTypes) to LPS-147481 Tweaks to accept setContentName and updateSelectedItem fun…

**Comment by cgmonte** [comment]
Just started reviewing :)

**Comment by cgmonte** [comment]
ci:test:sf

**Comment by cgmonte** [comment]
ci:test:relevant

**Comment by cgmonte** [comment]
ci:forward
